portality/bll/exceptions.py

```diff
@@ -66,6 +66,23 @@ class ArticleMergeConflict(Exception):
     """
     pass
 
+class IllegalStatusException(Exception):
+    """
+    Exception to raise when an application is in a state that is not allowed for the current action
+    """
+
+    def __init__(self, message=None):
```

Comment:
An exception with a message param is pretty retro in python3, but I'm aware this exists in plenty of other places in the DOAJ codebase.